source/first-boot.rst

```diff
@@ -1,5 +1,46 @@
 First-Boot Customization
 ========================
 
-This is a placeholder for more documentation about first-boot
-customization.
+First-boot customization is performed using ``cloud-init``.
+``cloud-init`` runs during boot of all Ubuntu instances, discovers the
+environment in which it is running and configures the instance
+accordingly.  As well as configuring networking and user access, it can
+also perform customization based on instructions provided by the cloud
+vendor (vendor data) and by users (user data).  We are specifically
+interested in vendor-data.
+
+Vendor Data
+-----------
+
+Configuring Vendor Data
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Vendor data is presented to the instance as a plain text file.  How
+this file is configured depends on the implementation of your cloud,
+so you will need to determine how to configure it for your cloud.  For
+the remainder of this document, the examples will be the text file that
+you should configure as vendor data.
+
+.. note::
+    For cloud implementations that don't support vendor-data from the
+    metadata service, the image can be modified to provide a similar
+    effect.  See :doc:`vendor-data-via-modification` for more details
+    on how this can be achieved.
+
+Testing Vendor Data
+~~~~~~~~~~~~~~~~~~~
+
+``cloud-init`` parses and process vendor-data and user-data in almost
+exactly the same way.  This means that you can easily test vendor-data
+that you're writing by passing it to an instance you're launching as
+user-data.
```

Comment:
"If user-data and vendor-data contain different settings for the same configuration item, user-data will over-ride vendor-data."